addons/account/models/account_move.py

```diff
@@ -2720,10 +2720,21 @@ def _get_computed_account(self):
         accounts = self.product_id.product_tmpl_id.get_product_accounts(fiscal_pos=fiscal_position)
         if self.move_id.is_sale_document(include_receipts=True):
             # Out invoice.
-            return accounts['income']
+            if accounts['income']:
+                account = accounts['income']
+            elif self.account_id:
+                account = self.account_id
+            else:
+                account = self.env['account.account']
```

Comment:
```
account = accounts['income'] or self.account_id
```
?
same below

Reply:
Agree, the shorter the better (I was following the style of the other get_computed_* )